app/controllers/task_controller.rb

```diff
@@ -37,10 +37,40 @@ def confirm_delete
     redirect_to :root
   end
 
+  def update
+    Task.update((params[:id]), completed_on: Time.now)
+
+    index
+  end
+
+  def uncheck_update
+    Task.update((params[:id]), completed_on:" ")
```

Comment:
It would be better to set this field to nil rather than an empty string - especially if you have to use this data for any additional logic later